gnovm/pkg/gnolang/frame.go

```diff
@@ -92,3 +99,133 @@ type Defer struct {
 	// a panic occurs and is decremented each time a panic is recovered.
 	PanicScope uint
 }
+
+type StacktraceCall struct {
+	Stmt  Stmt
+	Frame *Frame
+}
+type Stacktrace struct {
+	Calls           []StacktraceCall
+	NumFramesElided int
+}
+
+func (s Stacktrace) String() string {
+	var builder strings.Builder
+
+	for i := 0; i < len(s.Calls); i++ {
+		if s.NumFramesElided > 0 && i == maxStacktraceSize/2 {
+			fmt.Fprintf(&builder, "...%d frame(s) elided...\n", s.NumFramesElided)
+		}
+
+		call := s.Calls[i]
+		cx := call.Frame.Source.(*CallExpr)
+		switch {
+		case call.Frame.Func != nil && call.Frame.Func.IsNative():
+			fmt.Fprintf(&builder, "%s\n", toExprTrace(cx))
+			fmt.Fprintf(&builder, "    gonative:%s.%s\n", call.Frame.Func.NativePkg, call.Frame.Func.NativeName)
+		case call.Frame.Func != nil:
+			fmt.Fprintf(&builder, "%s\n", toExprTrace(cx))
+			fmt.Fprintf(&builder, "    %s/%s:%d\n", call.Frame.Func.PkgPath, call.Frame.Func.FileName, call.Stmt.GetLine())
+		default:
```

Comment:
```suggestion
		case call.Frame.GoFunc != nil:
```

and add a default case with `panic("StacktraceCall has a non-call Frame")`

Reply:
remember to add the default case

Reply:
yes normally added here: https://github.com/gnolang/gno/pull/2145/commits/6e30e4e949eb7c90f425cccc5032460091074829